rpc/client.go

```diff
@@ -191,6 +195,45 @@ func (c *Client) Healthy() <-chan struct{} {
 	return c.healthy.Load().(chan struct{})
 }
 
+func (c *Client) isHealthy() bool {
+	select {
+	case <-c.Healthy():
+		return true
+	default:
+	}
+	return false
+}
+
+// WaitHealthy blocks until the channel is healthy, a health indication has
+// been received or the HealthWait time has expired.
+func (c *Client) WaitHealthy() bool {
+	// If the channel is healthy, return immediately.
+	if c.isHealthy() {
+		return true
+	}
+
+	// If we shouldn't wait for healthy, return immediately. The healthReceived
+	// channel is closed after we receive the first health indication for the
+	// client (i.e. (un)successful heartbeat or failure to open the connection).
+	select {
+	case <-c.healthReceived:
+		return false
```

Comment:
This can race with the client becoming healthy: if we get `false` from `isHealthy` and it becomes healthy immediately afterwards, it will return false here. This should return `c.IsHealthy()`